pkg/api/types.go

```diff
@@ -820,6 +820,10 @@ type PodSpec struct {
 	// used must be specified.
 	// Optional: Default to false.
 	HostNetwork bool `json:"hostNetwork,omitempty"`
+	// PullSecrets is an optional list of secret names to use for pulling any of the images used by this PodSpec.
+	// If specified, these secrets will be passed to individual puller implementations for them to use.  For example,
+	// in the case of docker, only DockerConfig type secrets are honored.
+	PullSecrets []string `json:"pullSecrets,omitempty"`
```

Comment:
nit: PullSecretNames? The PullSecrets is a little bit misleading to me. Though I know we won't be lucky enough to get real secrets in plain json...